submissions/7SergSmith7/Tiny JS World/index.js

```diff
@@ -0,0 +1,105 @@
+/* Refer to https://github.com/OleksiyRudenko/a-tiny-JS-world for the task details
+   Complete the below for code reviewers' convenience:
+
+   Code repository: _put repo URL here_
+   Web app: _put project's github pages URL here_
+   */
+
+// ======== OBJECTS DEFINITIONS ========
+// Define your objects here
+class Habitant {
+  constructor(species, name, gender, phrase, friends = [], legs, props = []) {
+    this.species = species;
+    this.legs = legs;
+    this.name = name;
+    this.gender = gender;
+    this.phrase = phrase;
+    this.friends = friends;
+    this.addFriends(friends);
+    this.props = props;
+    this.addProps();
+  }
+  addProps() {
+    this.props = ["species", "legs", "name", "gender", "phrase"];
+  }
+  addFriends(friends) {
+    this.friends = [...this.friends, ...friends];
+  }
+
+  listFriends() {
+    return this.friends.length
+      ? this.friends.map((friend) => friend.name)
+      : "I am loner";
+  }
+
+  sayPhrase() {
+    return this.phrase;
+  }
+
+  toString() {
+    this.props = this.props.map((prop) => `<strong>${this[prop]}</strong>`);
```

Comment:
(1) When we call `toString` for any object for the 2nd time this part will break. 
Why? At the second call `this.prop` will contain smth like `['<strong>cat</strong>',....]` so while being mapped will fail at `this[prop]` as it would return `undefined` (as there is no property named `'<strong>cat</strong>'`

(2) This fragment still implicitly manages properties from child classes as it traverses `.props` array populated by a child class (the latter is basically legal though).

Solution:
List properties' names explicitly.
Manage strongization of only properties known to that particular class. You may want to use a helper function like ``function makeStringStrong(s) { return `<strong>${s}....` } `` to be used across all classes for consistency.